### ISSUE TITLE
SQL Server + gateway supported in AppOwnsData?

### ISSUE THREAD

**Issue by kdvis** [MicrosoftDocs/powerbi-docs#2966]
I am following the [tutorial](https://docs.microsoft.com/en-us/power-bi/developer/embedded/embed-paginated-reports-customers) for embedding a paginated report into an application for your customers. I understand that Power BI datasets and SSO datasources are not supported for AppOwnsData. Are SQL Server datasources (accessed through an on-prem gateway) supported for AppOwnsData? By following the documentation, it would appear as though they are not because embed token generation requires a dataset id, but datasets from SQL Server datasources at this time are not getting embedded when publishing a report to Power BI Online from Power BI Report Builder, so a dataset Id cannot be obtained.

I have created a [community post](https://community.powerbi.com/t5/Developer/Embedding-Paginated-Report-with-AppOwnsData-and-data-gateway/m-p/1772992) about this issue here. If there are specific instructions for AppOwnsData to access the SQL Server through the gateway, it should be mentioned in the tutorial guide.

**Comment by mihart** [comment]
https://docs.microsoft.com/en-us/power-bi/developer/embedded/embed-paginated-reports-customers

**Comment by KesemSharabi** [comment]
Hi @kdvis,

A paginated report connected to an on-prem SQL is supported. You don't need to provide a dataset ID for paginated reports.

[Create the embed token](https://docs.microsoft.com/power-bi/developer/embedded/embed-paginated-reports-customers#create-the-embed-token).

![image](https://user-images.githubusercontent.com/54834915/115234412-ae962400-a121-11eb-9ca4-06a158c8d411.png)

Hope this helps,

Kesem

**Comment by kdvis** [comment]
Thank you @KesemSharabi 

I think the code snippet you reference to is using a namespace from an older version of the Power BI .NET client library (Microsoft.PowerBI.Api.V2). If you download the most recent package from NuGet, it's now just Microsoft.PowerBI.Api and Microsoft.PowerBI.Api.Models; Microsoft.PowerBI.Api.V2 will give you an error. Using the most recent package, I also can't create an instance of EmbedConfig; it's identified as an unknown type.

The [AppOwnsData sample](https://github.com/microsoft/PowerBI-Developer-Samples/blob/master/.NET%20Core/Embed%20for%20your%20customers/AppOwnsData/AppOwnsData.csproj) available from github is using version 3.21
`<PackageReference Include="Microsoft.PowerBI.Api" Version="3.21.0" />`

For testing purposes, I pulled in a package reference to version 2.x, and then I could find this Microsoft.PowerBI.Api.V2 namespace (VS still wasn't recognizing EmbedConfig() though). So it seems as though the code snippet that appears here in this thread and in the tutorial guide are using an older version than what is used in the AppOwnsData sample. Is there a way to generate the embed configuration using the most recent Power BI Embedded library or do I have to revert to an older version?

**Comment by KesemSharabi** [comment]
Hi @kdvis,

You're right, the code snippets are old. We're in the process of updating all of our samples, and the  docs will follow once the samples are updated.

In any case, the new samples work with embed token V2 ([multi-resource](https://docs.microsoft.com/rest/api/power-bi/embedtoken/generatetoken)) which doesn't support paginated reports.

The AppOwnsData scenario supports paginated reports (using the report ID), but only for embed token V1 - so here you're right again - you'll have to use the old code (documented in the article). 

Thanks,

Kesem

**Comment by kdvis** [comment]
Thank you, @KesemSharabi , your guidance helped and my paginated report is embedding in my app! Here is what I did:

1. Downgraded Microsoft.PowerBI.Api version 3.x to version 2.x
2. Updated the section of the sample code that generates the embed token, using the code snippets from the docs as reference

I guess I will keep an eye out to see if and when V2 embed tokens will support paginated reports, but good to know I can get by for now by using V1.
